### DISCUSSION TITLE
10x  sector should be 1x collateral after verified deals expired

### DISCUSSION THREAD

**Discussion by flyworker** [filecoin-project/FIPs#367] [Enhancements - Technical]
We found that when we extend a 10x verified deal sector:

- The sector extend successfully
- 10x collateral is still locked in the extended sector
- The extended sector has normal raw power

Isn't it make more sense to unlock the 9x collateral to available balance?

**Comment by kaitlin-beegle** [comment]
Hey @flyworker, this is a great proposal.  Seems intuitive!

To my understanding- and please correct me if I'm wrong- there is nothing about extending a deal sector with verified client data that _requires_ a change in sector power.  If a sector is extended and a client still has datacap available, they can continue to pay for the sector with datacap, and the 10x power adjustment (and 10x collateral) is appropriate. 

The specification for this FIP, then, would need to adjust the collateral calculation to take into account the method of payment (i., datacap y/n) for the deal sector at extension.  Does this align with your thinking? 

Also, as an FYI, I've flagged this topic for the Filecoin+ folks as well :)

**Reply by anorth** [reply]
The current quality-adjusted power mechanism doesn't easily permit this kind of thing. The mechanism _does_ require that sector power be changed if the expiration is extended. "The extended sector has normal raw power" in the OP isn't quite correct: the sector's new power will apportion the quality boost from any remaining sector lifespan over the new lifespan. This may be _close_ to the raw power, but will be a bit higher (depending how long the sector has until initial expiration).

However, the post is really about the collateral requirement for this reduced power.

**Reply by steven004** [reply]
The solution is laid in FIP-0033 and #313 . 
The separation of raw-byte power calculation and deal-based collateral (CC sector could be considered as a special deal) could solve a lot things and provide lots of flexibilities. 

We may need to consider FIP-0033 be implemented sooner than later.

**Comment by flyworker** [comment]
@kaitlin-beegle  yes, you are correct, it is not about extending, it is about the collateral. The collateral should matches with the power

**Reply by kaitlin-beegle** [reply]
Sure, I understand! @flyworker are you able to put together a draft version of this FIP? Let me know if you'd like help/resources for doing so!

**Comment by Fatman13** [comment]
A side note, does snap deal take into account if a 10x collateral sector with expired datacap deal when snapping a datacap deal? Hopefully it doesn't require additional 10x collateral...

**Reply by kaitlin-beegle** [reply]
@Fatman13 is this an issue that you've experienced while trying to 'snap' sectors? If you are trying to 'snap' CC sectors, the deal you're upgrading your sectors with would be a new deal (to you, at least).  Seems like it would either be a verified client deal paid with datacap, or it wouldn't be.

**Reply by Fatman13** [reply]
It is not confirmed but a general question. If the way it works is like what @flyworker describes above, then after a datacap deal expired the 10x collateral would still be locked in the sector as opposed to 9x of it being released to available balance. If at this moment you try to snapdeal the said sector, not sure if the protocol would check the existing collateral or require additional collateral.

**Reply by anorth** [reply]
At present you cannot snap into a sector that has ever had a deal, only CC sectors. The complications of QA power accounting are the main reason for this limitation. When we resolve this through #313 and #298, the protocol will be aware and will not require _additional_ collateral if the existing stake is sufficient to secure the new deal.

**Comment by galen-mcandrew** [comment]
This is hitting some of the limits of my understanding about deal renewal. That said, I think verified renewals should be 10xQAP and 10x collateral. In my mind, there should be a way on-chain for a client to indicate that a deal is eligible for renewal (and they could mark that state when the deal is proposed, and update that state any time during the deal). This way a client can 'pre-approve' the renewal of verified deals.

**Reply by flyworker** [reply]
> This is hitting some of the limits of my understanding about deal renewal. That said, I think verified renewals should be 10xQAP and 10x collateral. In my mind, there should be a way on-chain for a client to indicate that a deal is eligible for renewal (and they could mark that state when the deal is proposed, and update that state any time during the deal). This way a client can 'pre-approve' the renewal of verified deals.

the pre-approve can be done via the FVM in the future, but now we need to fix the 1xQAP and 10x collateral after extend deal

**Reply by anorth** [reply]
Deal renewal is currently impossible. We will fix this through this through #313 and #298 (for FIL+). But @flyworker I understand this is not what you're talking about.

**Comment by anorth** [comment]
@flyworker you are correct that the network currently enforces a policy that a sector's collateral is monotonic: it can only increase. Even when we make sector storage much more flexible (deal extension, transferring deals, etc) the current plan is to maintain this collateral requirement at its high water mark.

I can certainly see why providers might wish for the collateral to be recalculated at various points (it almost always goes down, even if deals were extended). I've brought this up before, but forget the reasoning. We need the cryptoecon team to help explain why this policy is preferred. @zixuanzh @AxCortesCubero.

**Reply by AxCortesCubero** [reply]
I'm not sure I follow what the issue is. Currently collateral and QAP is calculated based on averaging over the sector lifetime. If a sector is sealed full of FIL+ deals, but for half of the sector lifetime, QAP and collateral will be 5X, for the whole duration of the sector (even after the deal expired halfway). @anorth 's proposals may change that, but that is the current status.

The monotonic aspect has always been about within one sector lifetime. The QAP will not decrease, because it was already calculated by averaging over all the sector lifetime, so it doesnt need to decrease after the deal ends, because the fact that the deal will end was already accounted for.

In any case in the future it could be allowed to add another FIL+ deal, which would only increase the QAP, with a bit more complicated accounting as you point out. But this is not possible at present.

The issue here is about renewing such a sector? If I understand this correctly, the claim is that if this sector is extended, and a new sector lifetime starts, it would be forced to keep the old collateral, yet have it QAP reduced to 1X because it no longer has FIL+? Is this the case?

**Reply by flyworker** [reply]
@AxCortesCubero  
`f I understand this correctly, the claim is that if this sector is extended, and a new sector lifetime starts, it would be forced to keep the old collateral, yet have it QAP reduced to 1X because it no longer has FIL+? Is this the case?`

Yes, you are correct. That's what I claimed

**Reply by NiwanDao** [reply]
Based on current situation, if I extend the sector right before the plus deal expires, will the sectors be CC again? That means, keeping the 10X QAP by Snapping deal.

**Reply by Fatman13** [reply]
> At present you cannot snap into a sector that has ever had a deal, only CC sectors. 

@xingjitansuo current implementation only allows CC to be snapped [once](https://github.com/filecoin-project/FIPs/discussions/367#discussioncomment-2787056) if I am not mistaken.

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
if you extend a sector with fil+ inside you will end up with a fil+ sectors that has an inactive deal inside and your power gets diluted over the whole lifetime of the extended sector.

so you end up with a strange sector that is neither CC nor fully FIL+ x10 but something in between. collateral should stay the same if i remember right

i totally agree with @flyworker that we need a clean solution for multipliers - with the current discussion going on about duration multipliers etc. - and the pledge locked up should always represent the point in time the last power adjustment was made -033 will make sure of that - i am sure

if a deal is active FIL+ it should give x10 power - if not then x1 and the sector/space needs to be considered CC again. resnapping, deal renewal, are essential tools that SPs need to do proper business dev.

**Reply by jennijuju** [reply]
#313 likely will resolve this

**Reply by steven004** [reply]
Yes, #313 and other related discussions to decouple the Fil+ and sectors will solve this. 
A clean solution needs to make sectors fungible, and we have deal based collateral perhaps, instead of calculating QAP.
